test/distributed/org/apache/cassandra/distributed/test/SystemKeyspacesDataLocationTest.java

```diff
@@ -0,0 +1,79 @@
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.junit.Test;
+
+import org.apache.cassandra.cql3.QueryHandler;
+import org.apache.cassandra.cql3.QueryProcessor;
+import org.apache.cassandra.db.ColumnFamilyStore;
+import org.apache.cassandra.db.SystemKeyspace;
+import org.apache.cassandra.distributed.Cluster;
+import org.apache.cassandra.schema.Schema;
+import org.apache.cassandra.schema.SchemaConstants;
+import org.apache.cassandra.schema.TableMetadata;
+
+import static com.datastax.driver.core.ParseUtils.doubleQuote;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SystemKeyspacesDataLocationTest extends TestBaseImpl
+{
+    @Test
+    public void preparedStatementsShouldNotBeMovedOnStartupTest() throws IOException, ExecutionException, InterruptedException
+    {
+        try (Cluster cluster = Cluster.build(1).withDataDirCount(3).start())
+        {
+            // to test the behaviour we need have sstables of prepared statements in more than one data directory
+            cluster.get(1).runOnInstance(() -> {
+                ColumnFamilyStore preparedStatementsCFS = ColumnFamilyStore.getIfExists(SchemaConstants.SYSTEM_KEYSPACE_NAME, SystemKeyspace.PREPARED_STATEMENTS);
+                preparedStatementsCFS.disableAutoCompaction();
+                ColumnFamilyStore.getIfExists(SchemaConstants.SYSTEM_KEYSPACE_NAME, SystemKeyspace.PREPARED_STATEMENTS).disableAutoCompaction();
+                for (String keyspaceName : SchemaConstants.LOCAL_SYSTEM_KEYSPACE_NAMES)
+                {
+                    for (TableMetadata tableMetadata : Schema.instance.getKeyspaceMetadata(keyspaceName).tables)
+                    {
+                        String query = String.format("select * from %s.%s", keyspaceName, doubleQuote(tableMetadata.name));
+                        QueryHandler.Prepared prepared = QueryProcessor.prepareInternal(query);
+                        QueryProcessor.storePreparedStatement(query, keyspaceName, prepared);
+                        preparedStatementsCFS.forceBlockingFlush();
+                    }
+                }
+            });
+            Function<Cluster, Set<String>> preparedStatementsDataLocations = c -> c.get(1).callOnInstance(() -> ColumnFamilyStore.getIfExists(SchemaConstants.SYSTEM_KEYSPACE_NAME, SystemKeyspace.PREPARED_STATEMENTS)
+                                                                                                                                 .getLiveSSTables().stream()
+                                                                                                                                 .map(sstr -> sstr.descriptor.directory.getParentFile().getParentFile().toString())
+                                                                                                                                 .collect(Collectors.toSet()));
```

Comment:
Using a method called `getPreparedStatementsDataLocations(Cluster)` would be better fro readability than a function.
It would be also good to document: ` sstr.descriptor.directory.getParentFile().getParentFile().toString()` so that people understand the logic behind it.